src/api.rs

```diff
@@ -80,20 +88,68 @@ impl API {
     };
   }
 
+  fn login() {
+    // Get credentials
+    let mut username = match env::var("CLINK_USERNAME") {
+      Ok(user) => {
+        if user.chars().all(char::is_alphanumeric) {
+          user
+        } else {
+          get_current_username().unwrap().into_string().unwrap()
+        }
+      }
+      Err(_) => get_current_username().unwrap().into_string().unwrap(),
+    };
+
+    println!(
+     "Please enter password for {}: ",
+     username
+    );
```

Comment:
**Line 93:** Is `$CLINK_USERNAME` supposed to be your CSH username? Are you expected to set it before entering this app? Personally, I'm not a fan of undocumented `.env` vars :upside_down_face: 

Have you considered $USER instead?

**Line 98 and Line 101:** This is redundant.

So you're looking at an environment variable, then if it exists and is alphanumeric (which I don't think is a guarantee, see below), you use it. If it's not valid, you get the current system username (probably a safe bet, but you never know), but then unwrap it a fuckton up here instead of waiting until you actually use it. And, you do the same thing if the environment variable doesn't exist in the first place.

![image](https://user-images.githubusercontent.com/42927786/157067015-e2a760ae-0d9e-46dd-abcf-1ae98552d69d.png)

I think you might be able to unwrap a `get_current_username` outside of this block.

Reply:
`$CLINK_USERNAME` is if I am using clink on my Mac or other device where my account name is not my CSH username. It's kinda like how SSH operates, either you give it a username to use in some form or default to the value of `$USER`, which is what `get_current_username` is doing.

Hmmmm…… the alphanumeric part was mostly to help prevent any injects, but after talking to @Mstrodl about how they can't break out since its no longer `sh -c` then I can remove that checker

Reply:
Works 4 me.